File: src/main/resources/output/IFC4X3_js/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcSolidModel/IfcSweptAreaSolid/IfcSweptAreaSolid.js
class IfcSweptAreaSolid extends IfcSolidModel {
    constructor() {
        /** @type {IFCPROFILEDEF} */
        this.SweptArea = null;
        /** @type {IFCAXIS2PLACEMENT3D} */
        this.Position = null;
    }

    // === EXTENDED BY ===
    // IfcDirectrixCurveSweptAreaSolid
    // IfcExtrudedAreaSolid
    // IfcRevolvedAreaSolid

    // === WHERE CLAUSES ===
    // SweptAreaType : SweptArea.ProfileType = IfcProfileTypeEnum.Area
}

File: output/ifc4x3_dev_73740fe4/entity/IfcSweptAreaSolid.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public abstract class IfcSweptAreaSolid extends IfcSolidModel {

    public IfcProfileDef SweptArea;
    public IfcAxis2Placement3D Position; // OPTIONAL
    // INVERSE attributes:
    // SweptAreaType : SweptArea.ProfileType = IfcProfileTypeEnum.Area;

    // SUPERTYPE OF:
    // - IfcDirectrixCurveSweptAreaSolid
    // - IfcExtrudedAreaSolid
    // - IfcRevolvedAreaSolid SUBTYPE OF IfcSolidModel

    // WHERE constraints:
    // SweptAreaType : SweptArea.ProfileType = IfcProfileTypeEnum.Area;
}
